File: server/features/notification.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// NotificationClearHistory describes the clear-notification-history RPC.
var NotificationClearHistory = featureregistry.Feature{
	ID:          "notification-clear-history",
	Title:       "Clear Notification History",
	Description: "Clears the notification history for the current user.",
	RPCIDs:      []string{"notification:clear-history"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// NotificationGetHistory describes the get-notification-history RPC.
var NotificationGetHistory = featureregistry.Feature{
	ID:          "notification-get-history",
	Title:       "Get Notification History",
	Description: "Retrieves the notification history for the current user.",
	RPCIDs:      []string{"notification:get-history"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// NotificationMarkRead describes the mark-notification-read RPC.
var NotificationMarkRead = featureregistry.Feature{
	ID:          "notification-mark-read",
	Title:       "Mark Notification Read",
	Description: "Marks one or more notifications as read by their IDs.",
	RPCIDs:      []string{"notification:mark-read"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// NotificationSend describes the send-notification RPC.
var NotificationSend = featureregistry.Feature{
	ID:          "notification-send",
	Title:       "Send Notification",
	Description: "Sends a push notification to subscribed clients and manages subscription lifecycle.",
	RPCIDs:      []string{"notification:send"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(NotificationClearHistory)
	featureregistry.Register(NotificationGetHistory)
	featureregistry.Register(NotificationMarkRead)
	featureregistry.Register(NotificationSend)
}

File: web-app/src/lib/features/features/notification.ts
import type { Feature } from '../types';

export const notificationFeatures = {
  'notification-clear-history': {
    id: 'notification-clear-history',
    title: 'Clear Notification History',
    description: 'Clears the notification history for the current user.',
    rpcIds: ['notification:clear-history'],
    componentPaths: [],
    testIds: ['TestClearNotificationHistory_Success'],
    status: 'stable',
    since: '1.0.0',
  },
  'notification-get-history': {
    id: 'notification-get-history',
    title: 'Get Notification History',
    description: 'Retrieves the notification history for the current user.',
    rpcIds: ['notification:get-history'],
    componentPaths: [],
    testIds: ['TestGetNotificationHistory_ReturnsEmpty'],
    status: 'stable',
    since: '1.0.0',
  },
  'notification-mark-read': {
    id: 'notification-mark-read',
    title: 'Mark Notification Read',
    description: 'Marks one or more notifications as read by their IDs.',
    rpcIds: ['notification:mark-read'],
    componentPaths: [],
    testIds: ['TestMarkNotificationRead_EmptyIDs'],
    status: 'stable',
    since: '1.0.0',
  },
  'notification-send': {
    id: 'notification-send',
    title: 'Send Notification',
    description: 'Sends a push notification to subscribed clients and manages subscription lifecycle.',
    rpcIds: ['notification:send'],
    componentPaths: [],
    testIds: [
      'TestSendNotification410RemovesSubscription',
      'TestSendNotification404RemovesSubscription',
      'TestSendNotificationClosesBody',
      'TestSendNotification201RetainsSubscription',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;
